src/client.ts

```diff
@@ -1131,31 +1154,26 @@ export class ZkBobClient {
       await this.updateState(tokenAddress);
     }
 
-
     const txFee = await this.atomicTxFee(tokenAddress);
-    const accountBalance = await state.accountBalance();
-    const notesParts = await this.getGroupedNotes(tokenAddress);
-
-    let summ = BigInt(0);
-    let oneTxPart = accountBalance;
-    let i = 0;
-    do {
-      if (i < notesParts.length) {
-        oneTxPart += notesParts[i];
-      }
+    const groupedNotesBalances = await this.getGroupedNotes(tokenAddress);
+    
+    let accountBalance = await state.accountBalance();
+    if (groupedNotesBalances.length == 0) {
+      return accountBalance > txFee ? accountBalance - txFee : BigInt(0);
+    }
 
-      if(oneTxPart < txFee) {
+    let maxAmount = BigInt(0);
```

Comment:
should it be this way?
```suggestion
    let maxAmount = accountBalance > txFee ? accountBalance - txFee : BigInt(0);
```

Reply:
Yes, it looks like we can simplify logic a bit. Done: [2d89f11](https://github.com/zkBob/zkbob-client-js/pull/89/commits/2d89f114b90b9c8f40ba150701192cc4cab18535)